File: src/main/java/com/empiricus/statusviajante/model/CategoriaGastoModel.java
package com.empiricus.statusviajante.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "categoria")
public class CategoriaGastoModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idCategoria;

    @NotNull
    @Size(min = 3, max = 50)
    private String nome;

    @OneToMany(mappedBy = "categoriaGastoModel")
    private Set<GastoViagemModel> gastosViagem = new HashSet<>();

    public Long getIdCategoria() {
        return idCategoria;
    }

    public void setIdCategoria(Long idCategoria) {
        this.idCategoria = idCategoria;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }
}

File: src/main/java/com/empiricus/statusviajante/model/CategoriaGastoModel.kt
package com.empiricus.statusviajante.model

import javax.persistence.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "categoria")
class CategoriaGastoModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idCategoria: Long? = null
    var nome: @NotNull @Size(min = 3, max = 50) String? = null


    @OneToMany(mappedBy = "categoriaViagem")
    private val gastosViagem: Set<GastoViagemModel> = HashSet()
}